projects/archiver/index.ts

```diff
@@ -0,0 +1,92 @@
+require('dotenv').config()
+
+import { Handler } from 'aws-lambda'
+import { getIssue, getFront } from './downloader'
+import { attempt, hasFailed, hasSucceeded, Attempt } from '../backend/utils/try'
+import { Front, issuePath, frontPath } from './common'
+import { upload } from './upload'
+import { zip } from './zipper'
+import { getImagesFromFront, uploadImage } from './media'
+import { unnest } from 'ramda'
+import { imageSizes } from '../common/src/index'
+
+const run = async (id: string): Promise<void> => {
+    console.log('Attempting to download issue.')
+    const issue = await attempt(getIssue(id))
+    if (hasFailed(issue)) throw new Error('OH NO')
+    console.log('Downloaded issue', id)
+    const maybeFronts = await Promise.all(
+        issue.fronts.map(
+            async (frontid): Promise<[string, Attempt<Front>]> => [
+                frontid,
+                await attempt(getFront(id, frontid)),
+            ],
+        ),
+    )
+    maybeFronts.forEach(([id, attempt]) => {
+        if (hasFailed(attempt)) {
+            console.warn(`Front ${id} failed with ${attempt.error}`)
+        }
+    })
+
+    console.log('Fetched fronts')
+
+    await Promise.all(
+        maybeFronts.map(async ([frontId, maybeFront]) => {
+            if (hasSucceeded(maybeFront)) {
+                return await attempt(
```

Comment:
I'm not quite familiar with `attempt` here but whatever it is we do with it we seem to be happy to throw the result away, is this ok? I'm guessing we might want to alert if this upload failed?